cloudinit/net/networkd.py

```diff
@@ -271,7 +299,26 @@ def _render_content(self, ns):
                             name = dev_name
                             break
                 if name in ns.config["ethernets"]:
-                    self.dhcp_domain(ns.config["ethernets"][name], cfg)
+                    device = ns.config["ethernets"][name]
+
+                    # dhcp{version}domain are extra keys only present in
+                    # VMware config
+                    self.dhcp_domain(device, cfg)
+                    for version in ["4", "6"]:
+                        if (
+                            f"dhcp{version}domain" in device
+                            and "use-domains"
+                            in device.get(f"dhcp{version}-overrides", {})
+                        ):
+                            exception = (
+                                f"{name} has both dhcp{version}domain"
+                                f" and dhcp{version}-overrides.use-domains"
+                                f" configured. Use one"
+                            )
+                            LOG.critical(exception)
```

Comment:
Nit: An exception raised from here will make it into the logs, so we don't need an additional log here too.

Reply:
Fixed. Thanks!